storage/innobase/lock/lock0lock.cc

```diff
@@ -7304,6 +7306,26 @@ and less modified rows. Bit 0 is used to prefer orig_trx in case of a tie.
         }
         snprintf(buf, sizeof buf, rollback_msg, victim_pos);
         print(buf);
+
+        if (srv_print_all_deadlocks)
+        {
+          long len= ftell(lock_latest_err_file);
+          if (len > 0)
+          {
+            deadlock_info= static_cast<char*>(ut_malloc_nokey(
+              static_cast<size_t>(len) + 1));
+            if (deadlock_info)
+            {
+              rewind(lock_latest_err_file);
+              deadlock_info_len= fread(deadlock_info, 1,
```

Comment:
Now the only place we need deadlock_info_len in -- is here. You can move its declaration here.